packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -64,11 +64,23 @@ abstract class KeyBase extends Resource implements IKey {
    */
   protected abstract readonly policy?: PolicyDocument;
 
+  /**
+   * Collection of aliases added to the key
+   *
+   * Tracked to determine whether or not the aliasName should be added to the end of its ID
+   */
+  private readonly aliases: Alias[] = [];
```

Comment:
This could be a counter, but I think the array is cleaner, and doesn't really impact resource usage

Reply:
It could also be a Boolean, but I am ok with this.